tests/functional/views/basic_lti_launch_test.py

```diff
@@ -137,45 +124,56 @@ def test_basic_lti_launch(
                 ),
             ),
             param(
-                "canvas_file_lti_params",
+                "canvas_file_launch_params",
                 "canvas_file_assignment",
                 id="canvas file launch with existing DB row",
             ),
             param(
-                "canvas_file_legacy_speedgrader_lti_params",
+                "canvas_file_legacy_speedgrader_launch_params",
                 None,
                 id="SpeedGrader launch with no existing DB rows",
             ),
             param(
-                "canvas_file_lti_params",
+                "canvas_file_launch_params",
                 None,
                 id="canvas file launch with no existing DB rows",
             ),
+            param(
+                "canvas_file_fixed_speedgrader_launch_params",
+                "canvas_file_assignment",
+                id="fixed SpeedGrader canvas file launch with existing DB row",
+            ),
+            param(
+                "canvas_file_fixed_speedgrader_launch_params",
+                None,
+                id="fixed SpeedGrader launch with no existing DB rows",
+            ),
+            param(
+                "canvas_file_fixed_speedgrader_launch_params",
+                "legacy_speedgrader_assignment",
+                id="fixed SpeedGrader launch with legacy SpeedGrader DB row",
+            ),
         ],
     )
     def test_canvas_file_assignment(
         self,
-        app,
+        post_launch,
         db_session,
-        lti_params_fixture_name,
+        launch_params_fixture_name,
         existing_assignment_fixture_name,
         request,
     ):
         if existing_assignment_fixture_name:
-            _ = request.getfixturevalue(existing_assignment_fixture_name)
+            request.getfixturevalue(existing_assignment_fixture_name)
 
-        lti_params = request.getfixturevalue(lti_params_fixture_name)
+        get_params, post_params = request.getfixturevalue(launch_params_fixture_name)
 
-        canvas_file_id = lti_params["file_id"]
-        canvas_course_id = lti_params["custom_canvas_course_id"]
+        canvas_file_id = post_params["file_id"]
+        canvas_course_id = post_params["custom_canvas_course_id"]
 
-        response = app.post(
-            "/lti_launches?learner_canvas_user_id=USER_ID",
```

Comment:
This was wrong and changing the launch type (always speedgrader) from the fixture.